src/rust/engine/fs/src/lib.rs

```diff
@@ -353,17 +353,67 @@ fn is_ignored<P: AsRef<Path>>(ignore: &Gitignore, path: P, is_dir: bool) -> bool
 }
 
 ///
+/// A wrapper around a CpuPool, to add the ability to drop the pool before forking,
```

Comment:
Can you explain in a comment here why dropping the pool before forking is desirable/necessary? And what will happen to any existing futures when this happens?

As far as I can tell, the pre_fork stuff is completely undocumented; can one of the pre_fork methods (in python, rust, C, or somewhere else) *also* have a nice comment explaining:
1. When things get forked
2. What is forking what
3. What guarantees we need which the pre_fork stuff provides

?